### PR TITLE
fix: [CDS-79153]: Using V2 feature flag enum instead of existing one

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
## Describe your changes

This pr does two things:

- As part of previous pr we had removed usages of CDS_QUERY_OPTIMIZATIONS as those were done quite a few months back. But I noticed these were enabled only on prod 2 globally. So I am reverting removal of those flags.
- All new changes I have created a new flag CDS_QUERY_OPTIMIZATIONS_V2 and put them behind this. This is so the changes are by default not enabled when the changes reach prod.

In future once some flag is enabled for a long time in prods we will move them to CDS_QUERY_OPTIMIZATIONS and new ones will go to V2

Hot fix pr for qa https://github.com/harness/harness-core/pull/53606

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53483)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/helpers/ext/account/DeleteAccountHelper.java [hunk 1]
```diff
@@ -181,7 +182,7 @@ List<String> deleteOwnedByAccountEntities(String accountId) {
 
   private boolean deleteAppLevelDocuments(String accountId, Class<? extends PersistentEntity> entry) {
     FindOptions findOptions = new FindOptions();
-    if (accountId != null && featureFlagService.isEnabled(FeatureName.CDS_QUERY_OPTIMIZATION, accountId)) {
+    if (accountId != null && featureFlagService.isEnabled(CDS_QUERY_OPTIMIZATION, accountId)) {
       findOptions.readPreference(ReadPreference.secondaryPreferred());
```
Comment (Abhishek-CDC) [inline 400-rest/src/main/java/software/wings/helpers/ext/account/DeleteAccountHelper.java:186]: how we are deciding v1 and v2 FF usage?

Comment (abhinavmittal-harness) [reply 400-rest/src/main/java/software/wings/helpers/ext/account/DeleteAccountHelper.java:186]: There were some optimisations that were done last week
Those are being moved to v2

**Review by rafaelmart1ns — APPROVED** [review]

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Review by lasdox — APPROVED** [review]

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java [hunk 2]
```diff
@@ -470,7 +471,9 @@ public PageResponse<StateMachine> listStateMachines(PageRequest<StateMachine> re
   @Override
   public List<Workflow> list(String accountId, List<String> projectFields, String queryHint) {
     FindOptions findOptions = new FindOptions();
-    findOptions.readPreference(ReadPreference.secondaryPreferred());
+    if (featureFlagService.isEnabled(CDS_QUERY_OPTIMIZATION, accountId)) {
```
Comment (sribalij) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:474]: Can we add some JUnits around the FFs? It'll help us when we remove or update them

**Review by sribalij — CHANGES_REQUESTED** [review]
Can we add some JUnits around the FFs? It'll help us when we remove or update them

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java (hunk 2 above)
Comment (sribalij) [reply 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:474]: Also a ticket to track the FF to remove

Comment (abhinavmittal-harness) [reply 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:474]: I will create a separate pr for that if it is okay. (As this change is just reverting a commit we did where we were removing this from code and we have decided not remove things from current gen)
Also Considering we want to hotfix changes to qa.
@sribalij

Comment (abhinavmittal-harness) [reply 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:474]: Also we won't be removing these feature flags

**Comment by abhinavmittal-harness** [comment]
trigger commonchecks

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java (hunk 2 above)
Comment (sribalij) [reply 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:474]: Sure. This sounds good @abhinavmittal-harness.

**Review by sribalij — APPROVED** [review]

**Review by naman-goenka — APPROVED** [review]

**Comment by abhinavmittal-harness** [comment]
trigger ut6

**Comment by abhinavmittal-harness** [comment]
trigger pmd

**Comment by abhinavmittal-harness** [comment]
trigger codebasehash

**Pull request merged**